lib/main.dart

```diff
@@ -1,117 +1,427 @@
-        // in the middle of the parent.
         child: Column(
-          // Column is also a layout widget. It takes a list of children and
-          // arranges them vertically. By default, it sizes itself to fit its
-          // children horizontally, and tries to be as tall as its parent.
-          //
-          // Invoke "debug painting" (press "p" in the console, choose the
-          // "Toggle Debug Paint" action from the Flutter Inspector in Android
-          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
-          // to see the wireframe for each widget.
-          //
-          // Column has various properties to control how it sizes itself and
-          // how it positions its children. Here we use mainAxisAlignment to
-          // center the children vertically; the main axis here is the vertical
-          // axis because Columns are vertical (the cross axis would be
-          // horizontal).
           mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text(
-              'You have pushed the button this many times:',
-            ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
-            ),
+          children: [
+            logined ? loginText : logoutText,
+            logined ? logoutBtnTwitter : loginBtnTwitter,
           ],
         ),
       ),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+
+  @override
+  _MyHomePageState createState() => new _MyHomePageState();
+
+  MyHomePage(FirebaseUser user, String token, String secret) {
+    _MyHomePageState.user = user;
+    _MyHomePageState.token = token;
+    _MyHomePageState.secret = secret;
+  }
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  static FirebaseUser user;
+  static String token;
+  static String secret;
+  static String deviceId;
+  static List<dynamic> certificateList = [];
+  static List<dynamic> fromNameList = [];
+  static List<dynamic> toNameList = [];
+  static List<dynamic> memoList = [];
+  static List<dynamic> dobList = [];
+
+
+  List<String> itemList = [];
+  void buildItemList()async{
+    var response = await fetchArticle();
+    for (int j = 0; j < response.data["items"].length; j++) {
+      setState(() {
+        itemList.add(response.data["items"][j]);
+      });
+    }
+  }
+
+  Future fetchArticle() async {
+
+    final url = "https://22161mw9kg.execute-api.ap-northeast-1.amazonaws.com/kosan_syoumei_mylist?device=$deviceId";
+
+    final response = await http.get(url);
+    if (response.statusCode == 200) {
+      setState(() {
+        certificateList = json.decode(response.body)["to_name_list"];
+        fromNameList = json.decode(response.body)["from_name_list"];
+        toNameList = json.decode(response.body)["to_name_list"];
+        memoList = json.decode(response.body)["memo_list"];
+        dobList = json.decode(response.body)["dob_list"];
```

Comment:
メンバー変数の数だけデコード処理が走ってしまうので、変数で受けてから参照するようにしたいです。

もしデコード処理ほど複雑ではなくても、同じメソッドを複数回呼ぶ場合は同様に対応したいです。
実装の詳細を知らなくても無駄なメソッドコールが無いことが分かります。

言語や環境によってはコンパイラが最適化してくれるかもしれませんが、いまの自分の知識ではどこまで最適化されるか分かりません。
コンパイラによる最適化の有無に関わらず最適化できるよう、記述を最適化した方がいいかなと思います。

```suggestion
      final body = json.decode(response.body)
      setState(() {
        certificateList = body["to_name_list"];
        fromNameList = body["from_name_list"];
        toNameList = body["to_name_list"];
        memoList = body["memo_list"];
        dobList = body["dob_list"];
```